src/ast/irbuilderbpf.cpp

```diff
@@ -2489,14 +2493,12 @@ void IRBuilderBPF::CreateHelperError(Value *return_value,
   CreateLifetimeEnd(buf);
 }
 
-// Report error if a return value < 0 (or return value == 0 if compare_zero is
-// true)
 void IRBuilderBPF::CreateHelperErrorCond(Value *return_value,
                                          libbpf::bpf_func_id func_id,
                                          const Location &loc,
-                                         bool compare_zero)
+                                         bool suppress_error)
```

Comment:
Rather than push suppression logic down into codegen, how about a new "unused return value" pass on AST? It'll

1. Get users in habit of checking return values
2. Avoid hyper-specific new flags on `Call` AST node
3. Keep codegen dumb

Reply:
I meant to say that we should stop emitting warnings from codegen for this error case completely. And then just create a pass that, for an allowlist of calls with important return values, warns if the return value is unused. Maybe these warnings can be silenced with a cast to void. I think it's less complicated and points users towards a direction we want to be going anyways.

> What specific new flags for Call nodes are you worried about?
```
bool ret_val_discarded = false;
```

Reply:
> This is a runtime warning not a compiler warning but maybe you're aware of that and saying we shouldn't issue a runtime warning even on failed deletes that don't handle the return value. If you are saying that then it's an interesting idea.

Right, I was suggesting delete the runtime error and make it strictly a compile time check.

> The part about that I don't love is just introducing even more warnings as all current scripts use delete as just an expression statement.

True, we are probably going to introduce a bunch of warnings to people's scripts. But like you said, perhaps it's important enough to do it? I'm not strictly opposed to the current approach in the PR - just surfacing a potential alternative.

Reply:
I was going to introduce a `update` method for maps so users can handle the return value, perhaps for that one we issue a warning if the return value is ignored (as a start).

Reply:
For the sake of terseness, I'd avoid printing a compile-time warning when the return value of `delete` is unused. For many scripts and one-liners, users will delete a key which always exists and they will want to use plain `delete(@map, X)`.

On top of that, I can see value in the runtime warning itself - many people will not realize that an update/delete may fail, would just suppress the compile-time warning, and then be surprised why the script is behaving strangely.